### ISSUE TITLE
Perhaps implement FizzBuzz in a more in-depth manner

### ISSUE THREAD

**Issue by zenware** [zenware/FizzBuzz#2]
http://www.tomdalling.com/blog/software-design/fizzbuzz-in-too-much-detail/

**Comment by sanspace** [comment]
May be there should be a separate folder or even a repository so you don't scare away beginners! :japanese_ogre:

**Comment by zenware** [comment]
Oh, I should hope beginners feel welcome to make mistakes, mistakes are 100% okay because this is not production and actually about learning.

Mistakes should be welcome anyways and I hope nobody would be mean, but there's a chance they will be. So... one just has to be comfortable making mistakes and being called out in the first place, whether in a mean or nice manner, and it also helps if your skin is thick enough or your esteem is high enough that some internet comments don't bother you that much.

I basically use this project as a quick and dirty reference manual for myself since it has 'enough syntax to get started' and then I can basically figure out from a compiler error or from some more detailed documentation what's going wrong. :)

**Comment by sanspace** [comment]
I was thinking along the lines of incremental complexity so they won't be overwhelmed. I see your point of being ready to commit mistakes and accept feedback too.

> it also helps if your skin is thick enough or your esteem is high enough that some internet comments don't bother you that much

Cannot agree more!

**Comment by zenware** [comment]
Wow here I am almost a year late and you get back to me within the hour, I'm really dropping the ball.

**Comment by sanspace** [comment]
Now, I remember. It was the time of [hacktoberfest](https://hacktoberfest.digitalocean.com/). That's why I was talking about making it easier for beginners. They were even adding hacktoberfest and beginnerfriendly labels to simple issues in some repos to welcome beginners to make their first PR possible.

**Comment by zenware** [comment]
Ah, interesting, part of this project is also like hunting down and configuring potentially esoteric compilers in the correct way, see #74 ...

So it's almost inherently not beginner friendly.
In the vein of making it more beginner friendly perhaps, everything could be more "folderized", and have a build system. That would also be an enabler for #17 

Maybe I'll look into something like that tonight.

**Comment by sanspace** [comment]
I understand. Glad that we had this conversation. Thank you.

**Comment by zenware** [comment]
@sanspace How timely, I opened the hacktoberfest page just before I commented the other day, and registered, and this morning it's going on! :+1: Definitely going to participate this seems fun

**Comment by sanspace** [comment]
Yeah, spread the word. Let's get more folks into opensource. I got my confidence after a hacktoberfest season. 😄

**Comment by zenware** [comment]
@sanspace add the one, carry the two, :scream: nooo, that means... your post a year ago was _because_ my project scared you off

**Comment by sanspace** [comment]
Nope. Luckily I bumped into your project on the second year of hacktoberfest! 😄
